contracts/Batcher.sol

```diff
@@ -71,12 +71,12 @@ contract Batcher {
     // Revert everything if any transfer fails
     for (uint8 i = 0; i < recipients.length; i++) {
       require(recipients[i] != address(0), 'Invalid recipient address');
+      emit BatchTransfer(msg.sender, recipients[i], values[i]);
```

Comment:
If the call function fails, the require(success, 'Send failed') statement will cause the transaction to revert. This will roll back the BatchTransfer event. And I think if the transaction reverts, the events and all state changes will also be reverted

Reply:
I think it's kind of a common pattern to prevent reentrancy attacks i.e. to make the state updates before the external call. If a state variable was updated after the call, a reentrant call could potentially interact with the contract again before the state was updated, leading to unexpected behavior.

Reply:
This pattern also applies to event emissions along with state updates